addressableled/robot.py

```diff
@@ -0,0 +1,51 @@
+import wpilib
+
+kLEDBuffer = 60
+
+
+class MyRobot(wpilib.TimedRobot):
+    def robotInit(self):
+        # PWM Port 9
+        # Must be a PWM header, not MXP or DIO
+        self.led = wpilib.AddressableLED(9)
+
+        # LED Data
+        self.ledData = [wpilib.AddressableLED.LEDData() for _ in range(kLEDBuffer)]
+
+        # Store what the last hue of the first pixel is
+        self.rainbowFirstPixelHue = 0
+
+        # Default to a length of 60, start empty output
+        # Length is expensive to set, so only set it once, then just update data
+        self.led.setLength(kLEDBuffer)
+
+        # Set the data
+        self.led.setData(self.ledData)
+        self.led.start()
+
+    def robotPeriodic(self):
+        # Fill the buffer with a rainbow
+        self.rainbow()
+
+        # Set the LEDs
+        self.led.setData(self.ledData)
+
+    def rainbow(self):
+        # For every pixel
+        for i in range(kLEDBuffer):
+            # Calculate the hue - hue is easier for rainbows because the color
+            # shape is a circle so only one value needs to precess
+            hue = (self.rainbowFirstPixelHue + (i * 180 / kLEDBuffer)) % 180
+
+            # Set the value
+            self.ledData[i].setHSV(int(hue), 255, 128)
+
+        # Increase by to make the rainbow "move"
+        self.rainbowFirstPixelHue += 3
+
+        # Check bounds
+        self.rainbowFirstPixelHue %= 100
```

Comment:
should this be 180 instead of 100?

Reply:
Good catch!

Reply:
thx @OliverW10
I didn't see this value :)